src/openrct2/drawing/X8DrawingEngine.cpp

```diff
@@ -406,19 +406,21 @@ void X8DrawingEngine::DrawAllDirtyBlocks()
             uint32_t yy;
             for (yy = y; yy < dirtyBlockRows; yy++)
             {
+                bool shouldBreak = false;
                 uint32_t yyOffset = yy * dirtyBlockColumns;
                 for (xx = x; xx < x + columns; xx++)
                 {
                     if (dirtyBlocks[yyOffset + xx] == 0)
                     {
-                        goto endRowCheck;
+                        uint32_t rows = yy - y;
+                        DrawDirtyBlocks(x, y, columns, rows);
+                        shouldBreak = true;
+                        break;
                     }
                 }
+                if(shouldBreak)
+                    break;
             }
-
-        endRowCheck:
-            uint32_t rows = yy - y;
```

Comment:
To fix @AaronVanGeffen 's comment, you have to restore these two lines here.

Reply:
Hmm, that wouldn't be the end of it, though. If the code `break`s above with these lines restored, `DrawDirtyBlocks` would be called twice. This can be avoided by using `return` instead of `break`, though, probably.